frontend/src/pages/admin/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { useContext } from "react";
+import {
+	Box,
+	Button,
+	Container,
+	Paper,
+	Table,
+	TableBody,
+	TableCell,
+	TableRow,
+	TableHead,
+	Typography,
+	CircularProgress,
+} from "@material-ui/core";
+import api from "../../helpers/api";
+import { AuthContext } from "../../context/auth";
+
+export default class AdminPage extends React.Component {
+	static contextType = AuthContext;
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			allUsersData: undefined,
+			selected: null
+		};
+	}
+
+	async componentDidMount() {
+		const { idToken, uid } = this.context.authState.user;
+		const allUsersData = (await api.admin.users.getAll(idToken)).data.data.users;
+		this.setState({ allUsersData });
+	}
+
+	handleRoleChange = (e, userId, roles) => {
+		const { idToken } = this.context.authState.user;
+		api.admin.users.roles.update(idToken, userId, roles)
+	}
+
+	handleUserDelete = (e, userId) => {
+		const { idToken } = this.context.authState.user;
+		api.admin.users.delete(idToken, userId);
+	}
```

Comment:
We can probably make this better by just using await on the API call, and if successful update the state with the change as well or we can just refresh the state completely with another call to update all the state for all the user data